src/astro/utils/load.py

```diff
@@ -58,28 +66,82 @@ def load_file_into_dataframe(
         elif filetype == FileType.JSON:
             dataframe = pd.read_json(stream, **kwargs)
         elif filetype == FileType.NDJSON:
-            dataframe = pd.read_json(stream, lines=True, **kwargs)
+            if database is None:
+                raise ValueError(
+                    "database param cannot be None when the file type is Ndjson."
+                )
+            dataframe = flatten_ndjson(
+                normalize_config, stream=stream, database=database
+            )
         elif filetype == FileType.PARQUET:
             dataframe = pd.read_parquet(stream, **kwargs)
         else:
             raise ValueError(f"Unable to load file '{filepath}' of type '{filetype}'")
         return dataframe
 
 
+def flatten_ndjson(
+    normalize_config: Union[None, dict], stream: io.TextIOWrapper, database: Database
+) -> pd.DataFrame:
```

Comment:
Can you add docstrings and explain the params too please

Reply:
My bad, missed it.